### ISSUE TITLE
Fine tuning whisper and whisper lora with prompts

### ISSUE THREAD

**Issue by kenfus** [huggingface/transformers#28549]
### Feature request

Currently, I am successfully able to fine-tune whisper and whisper Lora with timestamps, thank you for that!

Now, I would like to fine tune whisper with prompts, as Openai has trained it. There is not too much documentation, so maybe it's already in? Currently, my dataset has the following columns: `input_features`, `labels` and `prompt_ids`. The labels do currently not contain the `prompt_ids`. So my first question:

- If I add `prompt_ids` to the `labels` at the beginning, is it already correct? Will the huggingface library automatically cut the labels at the correct point and pass them to the model to start the decoding? I did not understand where this in the code exactly happens. 
- If not, where would it be best to add? I think either it should happen automatically from the `label` or we could make the trainer use `prompt_ids` automatically, if available. 

### Motivation

Overall, it was a bit unclear on how to finetune whisper with timestamps and prompts. Maybe it's already there, maybe not. In addition, the code was a bit hard to understand in the HU library.

### Your contribution

Absolutely! If I get some guidance, where to look and what to change, I am happy to do so. I am happy to contribute to HU, which helped me a lot in my work.

**Comment by amyeroberts** [comment]
Hi @kenfus, thanks for raising an issue! 

This is a question best placed in our [forums](https://discuss.huggingface.co/). We try to reserve the github issues for feature requests and bug reports.

cc @sanchit-gandhi @ylacombe

**Comment by kenfus** [comment]
Hey, thank you!

I have added a question to this post: https://discuss.huggingface.co/t/finetuning-whisper-with-prompts/43053/3

However, even when looking at the code, I don't quite understand how to pass the prompt to fine tune the model, so I think it's not implemented yet. With some guidance, I am very happy to implement it myself!

**Comment by amyeroberts** [comment]
Hi @kenfus, 

I'll let @sanchit-gandhi and @ylacombe answer about fine-tuning. For reference on the intended API, the PR to add prompts is #22395. There is also another open PR looking to add prompts to the pipeline #28556. Although these are not specific to the training regime, they may provide more context on how to pass prompts to the model.
